### DISCUSSION TITLE
Is it possible to output Users in Table by role condition?

### DISCUSSION THREAD

**Discussion by cleargoal** [filamentphp/filament#13448] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2

### How can we help you?

I created the UserResource and need to output to table users not with the role 'company'.
I tried such a query:
```
    public static function getEloquentQuery(): Builder
    {
        return parent::getEloquentQuery()->with('roles', function ($query) {
            $query->whereNot('name','company');
        });
    }
```

But it doesn't take effect.

What is the correct way for my purpose?

Any suggestions are appreciated.

**Comment by leandrocfe** [comment, marked as answer]
Use `modifyQueryUsing`
https://filamentphp.com/docs/3.x/panels/resources/listing-records#customizing-the-table-eloquent-query

**Reply by cleargoal** [reply]
Thank you!
I tried this method - look at the screenshot, please.
![image](https://github.com/filamentphp/filament/assets/10929782/526272c8-007d-45ea-8769-b2f17ca8af94)
But the result didn't change.

Can you explain please, how to debug the query inside the filament resource?

**Comment by cleargoal** [comment]
Finally, it works
```
        return $table
            ->modifyQueryUsing(fn(Builder $query) => $query->whereHas('roles', function (Builder $q) {
                $q->whereNot('name', 'company');
            }))
```

Thank you for the direction!
